plugin/core/sessions.py

```diff
@@ -966,7 +967,7 @@ def run_command(self, command: Mapping[str, Any]) -> Promise:
             )
         )
 
-    def run_code_action_async(self, code_action: Mapping[str, Any]) -> Promise:
+    def run_code_action_async(self, code_action: CodeAction) -> Promise[CodeAction]:
```

Comment:
The input should  be `Union[Command, CodeAction]`, the output should be `Promise[None]` I think?

Reply:
True about the return value. In the end the CodeAction is not returned.

The other one I have to check exactly

Reply:
> The input should be `Union[Command, CodeAction]`

Well, this is a nightmare to get working. Can't convince pyright that it's `Command` even with `isinstance` check or explicit `# type` comment.

![Screenshot 2020-11-16 at 21 29 37](https://user-images.githubusercontent.com/153197/99304617-e20f1280-2852-11eb-9cee-e813f42d46a0.png)

Reply:
I have to ignore this. Doesn't seem like pyright can infer the type of the parameter based on `isinstance` check of one of its properties.

Reply:
Used `cast` to fix it.